agimus_controller/tests/test_ocp_croco_base.py

```diff
@@ -0,0 +1,223 @@
+            frame_placement_residual = crocoddyl.ResidualModelFramePlacement(
+                self._state,
+                self._rmodel.getFrameId("panda_hand_tcp"),
+                pin.SE3(np.eye(3), np.array([1.0, 1.0, 1.0])),
+            )
+
+            goal_tracking_cost = crocoddyl.CostModelResidual(
+                self._state, frame_placement_residual
+            )
+            running_cost_model.addCost("stateReg", x_reg_cost, 0.1)
+            running_cost_model.addCost("ctrlRegGrav", u_reg_cost, 0.0001)
+            running_cost_model.addCost("gripperPoseRM", goal_tracking_cost, 1.0)
+            # Create Differential Action Model (DAM), i.e. continuous dynamics and cost functions
+            running_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+                self._state,
+                self._actuation,
+                running_cost_model,
+            )
+            running_model = crocoddyl.IntegratedActionModelEuler(
+                running_DAM,
+            )
+            running_model.differential.armature = self._robot_model.armature
+
+            running_model_list = [running_model] * (self._ocp_params.horizon_size - 1)
+            return running_model_list
+
+        def create_terminal_model(self):
+            # Terminal cost models
+            terminal_cost_model = crocoddyl.CostModelSum(self._state)
+
+            ### Creation of cost terms
+            ### Creation of cost terms
+            # State Regularization cost
+            x_residual = crocoddyl.ResidualModelState(
+                self._state,
+                np.concatenate((pin.neutral(self._rmodel), np.zeros(self._rmodel.nv))),
+            )
+            x_reg_cost = crocoddyl.CostModelResidual(self._state, x_residual)
+
+            # End effector frame cost
+            frame_placement_residual = crocoddyl.ResidualModelFramePlacement(
+                self._state,
+                self._rmodel.getFrameId("panda_hand_tcp"),
+                pin.SE3(np.eye(3), np.array([1.0, 1.0, 1.0])),
+            )
+
+            goal_tracking_cost = crocoddyl.CostModelResidual(
+                self._state, frame_placement_residual
+            )
+            terminal_cost_model.addCost("stateReg", x_reg_cost, 0.1)
+            terminal_cost_model.addCost("gripperPose", goal_tracking_cost, 50)
+
+            # Create Differential Action Model (DAM), i.e. continuous dynamics and cost functions
+            terminal_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+                self._state,
+                self._actuation,
+                terminal_cost_model,
+            )
+
+            terminal_model = crocoddyl.IntegratedActionModelEuler(terminal_DAM, 0.0)
+            terminal_model.differential.armature = self._robot_model.armature
+
+            return terminal_model
+
+        def set_reference_horizon(self, horizon_size):
+            ### Not implemented in this OCP example.
+            return None
+
+    def setUp(self):
```

Comment:
```suggestion
    def setUpClass(self):
```
I think you meant to use this. Otherwise before each and every test case the OCP will be solved